File: TOMIMSV4/Entity/Mantenimientos/Producto/ProductoEsadoUbic/clsBeProducto_estado_ubic.vb
<Serializable>
Public Class clsBeProducto_estado_ubic
    Implements ICloneable

    ''' <summary>
    ''' Gets or sets the identifier producto estad ubic.
    ''' </summary>
    ''' <value>The identifier producto estad ubic.</value>
    Public Property IdProductoEstadUbic() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier estado.
    ''' </summary>
    ''' <value>The identifier estado.</value>
    Public Property IdEstado() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier ubicacion defecto.
    ''' </summary>
    ''' <value>The identifier ubicacion defecto.</value>
    Public Property IdUbicacionDefecto() As Integer = 0
    ''' <summary>
    ''' Gets or sets the fec_agr.
    ''' </summary>
    ''' <value>The fec_agr.</value>
    Public Property Fec_agr() As Date = Date.Now
    ''' <summary>
    ''' Gets or sets the user_agr.
    ''' </summary>
    ''' <value>The user_agr.</value>
    Public Property User_agr() As String = ""
    ''' <summary>
    ''' Gets or sets the fec_mod.
    ''' </summary>
    ''' <value>The fec_mod.</value>
    Public Property Fec_mod() As Date = Date.Now
    ''' <summary>
    ''' Gets or sets the user_mod.
    ''' </summary>
    ''' <value>The user_mod.</value>
    Public Property User_mod() As String = ""
    ''' <summary>
    ''' Gets or sets a value indicating whether this <see cref="clsBeProducto_estado_ubic"/> is activo.
    ''' </summary>
    ''' <value><c>true</c> if activo; otherwise, <c>false</c>.</value>
    Public Property Activo() As Boolean = False

    ''' <summary>
    ''' Initializes a new instance of the <see cref="clsBeProducto_estado_ubic"/> class.
    ''' </summary>
    Sub New()
    End Sub

    ''' <summary>
    ''' Initializes a new instance of the <see cref="clsBeProducto_estado_ubic"/> class.
    ''' </summary>
    ''' <param name="idProductoEstadUbic">The identifier producto estad ubic.</param>
    ''' <param name="IdEstado">The identifier estado.</param>
    ''' <param name="IdUbicacionDefecto">The identifier ubicacion defecto.</param>
    ''' <param name="fec_agr">The fec_agr.</param>
    ''' <param name="user_agr">The user_agr.</param>
    ''' <param name="fec_mod">The fec_mod.</param>
    ''' <param name="user_mod">The user_mod.</param>
    ''' <param name="activo">if set to <c>true</c> [activo].</param>    
    Sub New(ByRef idProductoEstadUbic As Integer, ByVal IdEstado As Integer, ByVal IdUbicacionDefecto As Integer, ByVal fec_agr As Date, ByVal user_agr As String, ByVal fec_mod As Date, ByVal user_mod As String, ByVal activo As Boolean)
        Me.IdProductoEstadUbic = idProductoEstadUbic
        Me.IdEstado = IdEstado
        Me.IdUbicacionDefecto = IdUbicacionDefecto
        Me.Fec_agr = fec_agr
        Me.User_agr = user_agr
        Me.Fec_mod = fec_mod
        Me.User_mod = user_mod
        Me.Activo = activo
    End Sub
    ''' <summary>
    ''' Creates a new object that is a copy of the current instance.
    ''' </summary>
    ''' <returns>A new object that is a copy of this instance.</returns>
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function
End Class

File: WMS.EntityCore/Producto/clsBeProducto_estado_ubic.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMS.EntityCore.Producto
{
    public class clsBeProducto_estado_ubic : ICloneable
    {
        [Column("IdProductoEstadUbic")]
        [DisplayName("IdProductoEstadUbic")]
        public int IdProductoEstadUbic { get; set; } = 0;

        [Column("IdEstado")]
        [DisplayName("IdEstado")]
        public int IdEstado { get; set; } = 0;

        [Column("IdUbicacionDefecto")]
        [DisplayName("IdUbicacionDefecto")]
        public int IdUbicacionDefecto { get; set; } = 0;

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("IdBodega")]
        [DisplayName("IdBodega")]
        public int IdBodega { get; set; } = 0;

        public clsBeProducto_estado_ubic() { }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}
